notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+    "\n",
+    "The `uwtools` API's `config` module provides functions to create and manipulate configuration files, objects, and dictionaries.\n",
+    "\n",
+    "For more information, please see the <a href=\"https://uwtools.readthedocs.io/en/stable/sections/user_guide/api/config.html\">uwtools.api.config</a> Read the Docs page.\n",
+    "\n",
+    "## Table of Contents\n",
+    "\n",
+    "* [Getting Config Objects](#Getting-Config-Objects)\n",
+    "* [Config Depth Limitations](#Config-Depth-Limitations)\n",
+    "* [Realizing Configs](#Realizing-Configs)\n",
+    "    * [Updating configs](#Updating-configs)\n",
+    "    * [Using the `key_path` parameter](#Using-the-key_path-parameter)\n",
+    "    * [Using the `values_needed` parameter](#Using-the-values_needed-parameter)\n",
+    "    * [Using the `total` parameter](#Using-the-total-parameter)\n",
+    "* [Realizing Configs to a Dictionary](#Realizing-Configs-to-a-Dictionary)\n",
+    "<!--cell 0-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6d02d033-0992-4990-861d-3f80d09d7083",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "from uwtools.api import config\n",
+    "from uwtools.api.logging import use_uwtools_logger\n",
+    "\n",
+    "use_uwtools_logger()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "212594bb-379f-4441-805e-af0dbabe1815",
+   "metadata": {},
+   "source": [
+    "## Getting Config Objects\n",
+    "\n",
+    "The `config` tool can create configuration objects given a Python ``dict`` or one of five different file types: FieldTable, INI, Fortran namelist, Shell, or YAML. `config.get_yaml_config` is demonstrated here, but the config module also has similar functions for each of the other supported file types: `get_fieldtable_config()`,  `get_ini_config()`, `get_nml_config()`, and `get_sh_config()`.\n",
+    "<!--cell 2-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "ab0e21c3-a4b6-404c-bffd-e0d393d9b0a2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function get_yaml_config in module uwtools.api.config:\n",
+      "\n",
+      "get_yaml_config(config: Union[dict, str, pathlib.Path, NoneType] = None, stdin_ok: bool = False) -> uwtools.config.formats.yaml.YAMLConfig\n",
+      "    Get a ``YAMLConfig`` object.\n",
+      "\n",
+      "    :param config: YAML file or ``dict`` (``None`` => read ``stdin``).\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: An initialized ``YAMLConfig`` object.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.get_yaml_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "606da5b3-4bff-4148-a9a5-908aa7dd5e8c",
+   "metadata": {},
+   "source": [
+    "`get_yaml_config()` can take input from a Python `dict` or a YAML file like the one below.\n",
```

Comment:
```suggestion
    "The `stdin_ok` argument can be used to permit reads from `stdin`, but this is a rare use case beyond the scope of this notebook that will not be discussed here.\n",
    "\n",
    "`get_yaml_config()` can take input from a Python `dict` or a YAML file like the one below.\n",
```
I feel like we should mention this, if only to say that we're not going to mention it again.